force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -57,8 +65,16 @@ public with sharing class FlowControlLoopsCollections {
     public static Integer findMax(Integer num1, Integer num2) {
         // Initialize largestNum as null
         Integer largestNum = null;
-
-        return null; // Replace null with the variable you used to store the result
+        if (num1 ==null || num2 == null) {
+            return largestNum;
+        } else if (num1 == num2) {
+            largestNum = num1;
+            } else if (num1 > num2) {
+            largestNum = num1;
+              } else {
+                largestNum = num2;
+             }
+        return largestNum; 
```

Comment:
No need to check for equality on line 70. If the numbers are the same it does not matter which one you return. You could just use something like if(num1 >= num2) return .....

Take a look at the Math. max() method. You can save yourself a lot of code. 

https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_math.htm